src/ua_securechannel.c

```diff
@@ -510,25 +507,22 @@ sendChunkSymmetric(UA_ChunkInfo *ci, UA_Byte **buf_pos, const UA_Byte **buf_end)
         signature.length = securityPolicy->symmetricModule.cryptoModule.
             getLocalSignatureSize(securityPolicy, channel->channelContext);
         signature.data = *buf_pos;
-        ci->errorCode = securityPolicy->symmetricModule.cryptoModule.
+        ci->errorCode |= securityPolicy->symmetricModule.cryptoModule.
             sign(securityPolicy, channel->channelContext, &dataToSign, &signature);
-        if(ci->errorCode != UA_STATUSCODE_GOOD) {
```

Comment:
Why not check right after signing? Should we really try to encrypt a message if the signing process failed?

Reply:
I don't expect signing to fails a lot (if ever).
So we can save a bit in the hot-loop of every received message.
The error code from signing will be caught down the line.